packages/pipeline/src/scripts/scrape_data.ts

```diff
@@ -98,7 +106,29 @@ export const scrapeDataScripts = {
                 }
               });
         });
-    }
+    },
+    getHistoricalPriceSeries(fromSymbol: string,
+                             toSymbol: string,
+                             fromTimestamp: number,
+                             toTimestamp: number): any {
+        return new Promise((resolve, reject) => {
+            const limit = Math.min(
+                Math.round((toTimestamp - fromTimestamp) / (86400 * 1000)), 2000);
+            var parsedParams = querystring.stringify({
+                'fsym': fromSymbol,
+                'tsym': toSymbol,
+                'limit': limit,
+                'toTs': toTimestamp / 1000
+            });
+            request(HIST_PRICE_API_ENDPOINT + parsedParams, (error, response, body) => {
```

Comment:
Would it be possible to use `import * as request from 'request-promise-native';` here so that we can use `async/await` syntax rather then callbacks? Would make the code much easier to read. Happy to help with this as well.

Reply:
Refactored to use request-promise-native and async/await.

Reply:
Thanks for the help @fabioberger !